Docs 2146 business events v4.2 (#893)
mulesoft/docs-mule-runtime-3: modules/ROOT/pages/business-events-in-components.adoc

= Default Event Tracking
ifndef::env-site,env-github[]
include::_attributes.adoc[]
endif::[]

> Event tracking requires some processing and network overhead to aggregate and store the events that Mule runtime engine generates. By default, tracking is disabled for connectors or message processors that support it.

Reviewer: This text is titled "default tracking," but the previous sentence just staid that tracking is disabled? Why is it disabled if it is necessary?
So...should this text really say...
```suggestion
Because event tracking requires processing and network overhead to aggregate and store the events that Mule runtime engine generates, it is disabled by default. However, you can enable and configure default event tracking for connectors or message processors that support it, at one of two levels:
```

> Default event tracking can be configured in two ways:

Reviewer: OMIT per incorporation into previous.

> * At the flow level.

Reviewer: ```suggestion
* At the flow level
```

> * At the message processor (connector or component) level, which takes precedence over the flow level setting.

Reviewer: ```suggestion
* At the message processor (connector or component) level, which takes precedence over the flow level setting
```

Reviewer: This topic set intro text should also mention the third subtopic within, which is customizing the transaction ID.

[events_tracking_flow_level]

> == Event Tracking at The Flow Level

Reviewer: ```suggestion
== Event Tracking at the Flow Level
```
OR
Flow-Level Event Tracking

Use this option to enable default event tracking for all elements in a flow that support event tracking.

> You can disable tracking for specific processors or connectors to override the flow level setting.

Reviewer: I am a little confused about why there are two methods of enabling event tracking, because it seems to me as if they both accomplish the same thing:
Flow=enable all elements that support tracking; disable individual others as desired?
Processor=enable only individual elements that support?
If you think your audience might be confused a little, can you clarify this a bit further?
Maybe at the beginning of each task, state specifically WHY the user would choose one method over the other?

> Steps to enable:

Reviewer: ```suggestion
Steps to enable all supported event tracking:
```

. Select your flow component to open the properties view.
. Enable default business events using any of the following options:
+

> * In the UI, select *Enable default events tracking*.

Reviewer: ```suggestion
* In the UI, select *Enable default events tracking*:
```

+
image::mruntime-business-event-flow.png[default-event-tracking-flow]
+

> * In the XML, add attribute `tracking:enable-default-events="true"` to the `flow` element, for example:

Reviewer: ```suggestion
* In the XML, add attribute `tracking:enable-default-events="true"` to the `flow` element:
```

+
[source,xml,linenums]
----
<flow name="testFlow" tracking:enable-default-events="true">
----

> [event_tracking_processor_level]

Reviewer: Does the user not have to save these settings? Does the task end with one of the previous actions?

> == Enabling Event Tracking at The Message Processor Level

Reviewer: ```suggestion
== Enabling Event Tracking at the Message Processor Level
```
OR
Message Processor-Level Event Tracking

> You can enable event tracking on individual connectors and components that support event tracking. Not all processors support default event tracking. If the *Enable default events tracking* option is not present in a message processor or connector, default tracking is not supported.

Reviewer: See edit suggestion at the beginning of the other method. How is enabling individual connectors and components different from disabling some, results-wise? 
Also, I don't think we need to repeat that not all processors support tracking, or that it is not supported if the option is not enabled....

> Steps to enable:

Reviewer: ```suggestion
Steps to enable only some default event tracking:
```
Is this rewording correct?

> . Open the Connector or component properties view.

Reviewer: ```suggestion
. Open the connector or component properties view.
```

> . Enable default business events using any of the following options:

Reviewer: DITTO: default business event tracking?
"using either the UI or XML:"

+

> * In the UI, select *Enable default events tracking* inside the business events panel.

Reviewer: ```suggestion
* In the UI business events panel, select *Enable default events tracking*:
```

+
image::mruntime-business-event-component.png[default-event-tracking-component]
+

> * In the XML, add attribute `tracking:enable-default-events="true"` inside your component element, for example:

Reviewer: ```suggestion
* In the XML, add attribute `tracking:enable-default-events="true"` inside your component element:
```

= Configure Default Events Tracking
ifndef::env-site,env-github[]
include::_attributes.adoc[]
endif::[]

Because event tracking requires processing and network overhead to aggregate and store the events that Mule runtime engine generates, it is disabled by default. However, you can enable and configure default events tracking for connectors or message processors that support it, at one of two levels:

* At the flow level
* At the message processor (connector or component) level, which takes precedence over the flow level setting

After you enable event tracking, customize the transaction ID to identify specific tracked events so that you can analyze them at runtime.

[events_tracking_flow_level]
== Flow-Level Event Tracking

Use this option to enable default event tracking for all elements in a flow that support event tracking.
You can disable tracking for specific processors or connectors to override the flow level setting.

> Steps to enable default events tracking for all components in a Flow:

Reviewer 2: in a Flow: -> in a flow:

  ...
</flow>
----
. Save your settings.

[events_transaction_id]

> == Customizing the Transaction ID

Reviewer 2: Customizing the Transaction ID > Customize the Transaction ID

[events_transaction_id]
== Customizing the Transaction ID

> Define a transaction ID so that meaningful information, such as an order number, is displayed for a transaction when you analyze tracked events at runtime. If you do not customize the transaction ID, Mule assigns a numeric transaction ID by default.

Reviewer: Consider adding some text to connect this topic with the previous two, something like "After you enable event tracking, you need a way to identify specific tracked events so that you can analyze them at runtime."
See earlier comment suggesting that this become part of the topic set introduction.

> The transaction ID supports DataWeave expressions. Each custom ID must be unique for each transaction in your application. Follow these steps to set a transaction ID:

Reviewer: What is the importance of the ID "supporting" DataWeave expressions? 
And do we really need to state that an ID must be unique?
Finally, I suggest, again, introducing the steps by naming the two methods: "...to set a transaction ID either in the UI or in XML:"
```suggestion
The transaction ID supports DataWeave expressions. Each custom ID must be unique for each transaction in your application. Follow these steps to set a transaction ID:
```

* In the UI:
+
. Drag the *Set Transaction Id* component to your flow.

> . Define a transaction ID.

Reviewer: ```suggestion
. Define a transaction ID:
```

+
image:mruntime-custom-transaction-id.png[custom-transaction-id]

> * In the XML, add a child element to the `flow` element, for example:

Reviewer: ```suggestion
* In the XML, add a child element to the `flow` element:
```
